internal/distro/rhel82/distro_test.go

```diff
@@ -93,3 +96,321 @@ func TestFilenameFromType(t *testing.T) {
 		})
 	}
 }
+
+func TestImageType_BuildPackages(t *testing.T) {
+	x8664BuildPackages := []string{
+		"dnf",
+		"dosfstools",
+		"e2fsprogs",
+		"grub2-pc",
+		"policycoreutils",
+		"qemu-img",
+		"systemd",
+		"tar",
+		"xz",
+	}
+	aarch64BuildPackages := []string{
+		"dnf",
+		"dosfstools",
+		"e2fsprogs",
+		"policycoreutils",
+		"qemu-img",
+		"systemd",
+		"tar",
+		"xz",
+	}
+	buildPackages := map[string][]string{
+		"x86_64":  x8664BuildPackages,
+		"aarch64": aarch64BuildPackages,
+	}
+	d := rhel82.New()
+	for _, archLabel := range d.ListArches() {
+		archStruct, err := d.GetArch(archLabel)
+		if assert.NoErrorf(t, err, "d.GetArch(%v) returned err = %v; expected nil", archLabel, err) {
+			continue
+		}
+		for _, itLabel := range archStruct.ListImageTypes() {
+			itStruct, err := archStruct.GetImageType(itLabel)
+			if assert.NoErrorf(t, err, "d.GetArch(%v) returned err = %v; expected nil", archLabel, err) {
+				continue
+			}
+			reflect.DeepEqual(itStruct.BuildPackages(), buildPackages[archLabel])
```

Comment:
Why do we need `DeepEqual` instead of `assert.Equal` ? What are we trying to validate here that assert isn't capable of ?

Reply:
Note2: as-is `DeepEqual` will return a boolean and we don't really care about the value so in case of failure we'll not know about it.

Reply:
There was no explicit need for `DeepEqual` IMO, it was just a part of a previously existing code I reused (which was, AFAIK, based on a pre-generated boilerplate code). I can change it to `assert.Equal`, however in such a case I should also change other occurrences in the code that has already been merged, for the sake of consistency (in a separate PR). The second note seems to have a bigger impact, thanks for pointing this out.

Reply:
+1 for using `asserts.Equal` if we don't need DeepEqual. After you make the related changes maybe we can drop that package altogether.